### ISSUE TITLE
What's the real reason docker can't be used in China?  docker 在中国不能用的真正原因到底是什么？

### ISSUE THREAD

**Issue by travisbikkle** [docker/cli#5373]
### Description

What's the real reason docker can't be used in China? docker
在中国不能用的真正原因到底是什么？

What I've read in China is that docker is officially not allowed to be used by companies on the entity list, but none of the official releases have come out. So is this statement true? What is the url of this docker policy? 
我在中国看到的消息是docker官方不允许美国列入实体清单的公司使用，但是没有一个官方发文出来。所以这个说法是真的吗？
这个docker的政策网址在哪里？

Also why would the ce version of docker be affected? Does open source have national boundaries?
另外ce版本的docker为什么会受到影响？开源有国界？

Is it that docker maintainers just don't want Chinese people to use it? If that's the case I can live with it.
是docker就是不想让中国人用？如果是这样我可以接受，我不用就是了。

Or are Chinese companies and governments looking to get out of control and just want to get a whole new set out on their own?
还是中国公司和政府希望摆脱控制，就是希望自己弄一套全新的东西出来？

Let's all let go of our preconceived notions and tell us what the reason really is?
大家放开成见，说说看原因真正是什么？

### Reproduce

it not a bug

### Expected behavior

it not a bug

### docker version

```bash
it not a bug
```

### docker info

```bash
it not a bug
```

**Comment by thaJeztah** [comment]
This issue is not related to the CLI itself, and would be more suitable for https://github.com/docker/hub-feedback, but there's an existing issue (more below);

- https://github.com/docker/hub-feedback/issues/2388


That said, I'm not sure if there's much that can be done; I know there's been cases where specific political regions were blocked for legal reasons (Docker being a USA company, but I'm not a lawyer) but, at least to my knowledge, Docker itself does not deny access from China. I know sometimes access has been blocked by the GFW ([Great Firewall](https://en.wikipedia.org/wiki/Great_Firewall)), and there are some (unofficial) mirrors of Docker Hub in China for that reason (I think AliCloud still has a mirror, possibly others). There's also been some incidents where DNS was involved; see https://github.com/docker/hub-feedback/issues/2402#issuecomment-2243452691 and (more details);  https://github.com/docker/hub-feedback/issues/2388

> Docker has no intention of blocking access to Docker Hub from China. We have found that many users in China have incorrect DNS records for our servers, or that they are encountering some other sort of network issue between their systems and our servers.


I'm closing this ticket because of the above, and because it's a duplicate of https://github.com/docker/hub-feedback/issues/2388

**Comment by UjuiUjuMandan** [comment]
No, your patriotism is disgusting, fxxx off your entity list; just the Chinese Communist Party doesn't want you to download Xi Jinping speech synthesizer (习近平讲话模拟器) from Docker Hub.

https://www.google.com/search?q=习近平+Docker

You can do some research yourself, compare the blocking date and when they started to promote that image.

> [...It’s hard to comment, after all, it’s really not good if it gets out.](https://archive.today/2025.06.12-160526/https://nitter.net/PixelGeek606/status/1830522312044753201)

Just another Chinese taboo cannot be mentioned, and better to live in your China dream and think it's US sanctions.

**Comment by woodyxin** [comment]
github npm lol
